lib/train/file/remote/unix.rb

```diff
@@ -22,8 +22,13 @@ def content
         def exist?
           @exist ||= begin
             f = @follow_symlink ? "" : " || test -L #{@spath}"
-            @backend.run_command("test -e #{@spath}" + f)
-              .exit_status == 0
+            if @backend.run_command("uname -s").stdout.chomp == "SunOS"
```

Comment:
I don't know train very well, but I wonder if the current target family/arch/version are already detected at this point and are accessible to this class.... that would save the round-trip on an additional command call every time.

Reply:
This was the part that I spent the most time diving on I think. I hooked in to existing variables/methods, then looked around where you can inspect for version/etc but they weren't available here. I kept thinking *there must be a better way*. 

I'll try to pry back in and show off my instance vars and methods available here.

Reply:
Quick check with a pry:

```ruby
results_vars = {}
instance_variables.each do |k|
   results_vars[k] = eval k.to_s
end 

#=> {:@backend=>Train::Transports::SSH::Connection[unknown],
 :@path=>"/etc/passwd",
 :@follow_symlink=>true,
 :@spath=>"/etc/passwd",
 :@stat=>{:type=>:file, :mode=>420, :owner=>"root", :uid=>0, :group=>"sys", :gid=>3, :mtime=>1589990200, :size=>759, :selinux_label=>nil},
 :@mounted=>#<struct Train::Extras::CommandResult stdout="", stderr="", exit_status=1>,
 :@exist=>true,
 :@content=>
  "root:x:0:0:Super-User:/:/sbin/sh\ndaemon:x:1:1::/:\nbin:x:2:2::/usr/bin:\nsys:x:3:3::/:\nadm:x:4:4:Admin:/var/adm:\nlp:x:71:8:Line Printer Admin:/usr/spool/lp:\nuucp:x:5:5:uucp Admin:/usr/lib/uucp:\nnuucp:x:9:9:uucp Admin:/var/spool/uucppublic:/usr/lib/uucp/uucico\nsmmsp:x:25:25:SendMail Message Submission Program:/:\nlisten:x:37:4:Network Admin:/usr/net/nls:\ngdm:x:50:50:GDM Reserved UID:/:\nwebservd:x:80:80:WebServer Reserved UID:/:\npostgres:x:90:90:PostgreSQL Reserved UID:/:/usr/bin/pfksh\nsvctag:x:95:12:Service Tag UID:/:\nunknown:x:96:96:Unknown Remote UID:/:\nnobody:x:60001:60001:NFS Anonymous Access User:/:\nnoaccess:x:60002:60002:No Access User:/:\nnobody4:x:65534:65534:SunOS 4.x NFS Anonymous Access User:/:\nvagrant:x:100:100::/export/home/vagrant:/sbin/sh\n"}

results = {}
public_methods(false).each do |k|
   results[k] = send(k) rescue nil # in case args required
end  

#=> {:link_path=>nil,
 :path=>"/etc/passwd",
 :mounted=>#<struct Train::Extras::CommandResult stdout="", stderr="", exit_status=1>,
 :unix_mode_mask=>nil,
 :linked_to?=>nil,
 :content=>
  "root:x:0:0:Super-User:/:/sbin/sh\ndaemon:x:1:1::/:\nbin:x:2:2::/usr/bin:\nsys:x:3:3::/:\nadm:x:4:4:Admin:/var/adm:\nlp:x:71:8:Line Printer Admin:/usr/spool/lp:\nuucp:x:5:5:uucp Admin:/usr/lib/uucp:\nnuucp:x:9:9:uucp Admin:/var/spool/uucppublic:/usr/lib/uucp/uucico\nsmmsp:x:25:25:SendMail Message Submission Program:/:\nlisten:x:37:4:Network Admin:/usr/net/nls:\ngdm:x:50:50:GDM Reserved UID:/:\nwebservd:x:80:80:WebServer Reserved UID:/:\npostgres:x:90:90:PostgreSQL Reserved UID:/:/usr/bin/pfksh\nsvctag:x:95:12:Service Tag UID:/:\nunknown:x:96:96:Unknown Remote UID:/:\nnobody:x:60001:60001:NFS Anonymous Access User:/:\nnoaccess:x:60002:60002:No Access User:/:\nnobody4:x:65534:65534:SunOS 4.x NFS Anonymous Access User:/:\nvagrant:x:100:100::/export/home/vagrant:/sbin/sh\n",
 :mode=>420,
 :group=>"sys",
 :uid=>0,
 :gid=>3,
 :mode?=>nil,
 :sanitize_filename=>nil,
 :selinux_label=>nil,
 :size=>759,
 :mtime=>1589990200,
 :shallow_link_path=>nil,
 :type=>:file,
 :owner=>"root",
 :exist?=>true,
 :grouped_into?=>nil}
```

Reply:
If `@backend` is available here then `@backend.platform` will have all the detected platform details

Reply:
Aha - yes. With a tiny look I found `@backend.platform.solaris?` which is *much* cleaner - thanks!